### PR TITLE
select the fileid first when looking for incomplete files

### PR DESCRIPTION
**Pull request by icewind1991**
this seems to improve mariadb's index selection

### CODE REVIEW

**Review by come-nc — APPROVED** [review]

File: lib/private/Files/Cache/Cache.php [hunk 1]
```diff
@@ -1012,24 +1012,32 @@ public function getAll() {
...
 		}
 
-		// Make sure Oracle does not continue with null for empty strings
-		return (string)$path;
+		$path = $this->getPathById($id);
+		if ($path === null) {
+			return false;
+		} else {
+			return $path;
+		}
```
Comment (Altahrim) [inline lib/private/Files/Cache/Cache.php:1040]: Could be shorter: 
```php
return $path ?? false;
```

**Review by Altahrim — APPROVED** [review]

**Comment by solracsf** [comment]
@AndyScherzinger no 25 ?

**Comment by AndyScherzinger** [comment]
2023-08-15 10_00_41-select the fileid first when looking for incomplete files by icewind1991 · Pull

@solracsf ... srsly... ^^

**Pull request merged**